full/nvm-lazy.sh

```diff
@@ -1,18 +1,36 @@
 #!/bin/bash
 
-# Lazy-load NVM
-# Based on https://www.reddit.com/r/node/comments/4tg5jg/lazy_load_nvm_for_faster_shell_start/d5ib9fs
+lazy_load_nvm() {
+    unset -f node npm npx yarn
+    export NVM_DIR="$HOME/.nvm"
+    # shellcheck disable=SC1091,SC1090
+    [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh" # This loads nvm
+    # shellcheck disable=SC1091,SC1090
+    [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
+}
 
-declare -a NODE_GLOBALS=(`find ~/.nvm/versions/node -maxdepth 3 -type l -wholename '*/bin/*' | xargs -n1 basename | sort | uniq`)
```

Comment:
FYI, if I run `find ~/.nvm/versions/node -maxdepth 3 -type l -wholename '*/bin/*' | xargs -n1 basename | sort | uniq` in a regular Gitpod workspace, I find all these:

```
node-gyp
npm
npx
tsc
tsserver
yarn
yarnpkg
```

Not necessary a deal-breaker, but why stop searching for binaries under `~/.nvm/versions/node`?

Reply:
@jankeromnes good point. Please check the last version.

What we really need is to lazy load nvm itself, not any of the commands.